src/languageservice/services/yamlCompletion.ts

```diff
@@ -313,8 +313,20 @@ export class YamlCompletion {
         }
       }
 
-      const originalNode = node;
+      let originalNode = node;
       if (node) {
+        // when the array item value is null but the cursor is between '-' and null value (cursor is not at the end of the line)
+        if (isSeq(node) && node.items.length && isScalar(node.items[0]) && lineContent.includes('-')) {
+          const nullNode = node.items[0];
+          if (
+            nullNode.value === null && // value is null
+            nullNode.range[0] > offset // cursor is before null
+          ) {
+            node = nullNode;
+            originalNode = node;
+            overwriteRange.end.character += nullNode.range[2] - offset; // extend range to the end of the null element
```

Comment:
I do not think this is going to be enough to insert a valid yaml. The `overwriteRange.start.character` should be adjusted so that there is no extra space between `-` and the inserted value.

Reply:
thanks, I'll have a look

Reply:
Do you mean this one?

https://user-images.githubusercontent.com/38421337/162406730-3a49ed3e-1cac-41bc-8b71-a645addc038a.mov

hard to say, it looks ugly (extra indent), but the yaml is valid I think

Reply:
Yes, in my tests when I inject a single item editor marks it as invalid.